jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -275,10 +275,13 @@ YUI.add('juju-env-base', function(Y) {
 
       @method close
     */
-    close: function() {
+    close: function(cb) {
       if (this.ws) {
         this.beforeClose(this.ws.close.bind(this.ws));
       }
+      if (cb) {
+        cb();
```

Comment:
`on_close` sets the connected state of the env to false.
`connect` however creates a new websocket and I'm not totally sure that that's what we want.